src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -285,17 +313,45 @@ abstract class AbstractServerTask extends AbstractTask {
 
         // envProjectProps and serverEnvFile take precedence over server.env from configDirectory
         File envFile = new File(serverDirectory, "server.env")
+        Properties serverEnvProps = convertServerEnvToProperties(envFile);
+        
         if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
-            if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is overwritten by inlined configuration.")
+            Properties configuredProps = combineServerEnvProperties(server.env, envProjectProps);
```

Comment:
So I wasn't sure where to talk about this, but with the way this code is written it appends server env properties from multiple sources except certain combinations....which I think will be confusing for the customer. I'm going to try to capture this in this comment, but may need to create a table outside of here and attach an image later.

The first section of code dealing with the config directory will append the server.env from configDirectory with the default server.env. Then this code here will append the inline server env variables (from server.env Properties or project properties) with the potentially already appended server.env on the target server. If they also specified a serverEnvFile, that will not get appended because we said the inline configuration takes precedence. But in this appending scenario I think that is confusing.

Finally, if they did not specify inline configuration (through server.env Properties or project Properties) but did specify serverEnvFile, then that will get appended to the server.env in the target server.

I think we either need to append all configured server.env properties (no matter where they came from), or only the configured ones that previously took precedence. The previous precedence was (with 1 being the highest precedence):

1. project properties
2. server.env properties
3. serverEnvFile
4. server.env from configDirectory